src/libtrx/game/anims/frames.c

```diff
@@ -5,11 +5,17 @@
 #include "game/objects/common.h"
 #include "log.h"
 
+#define EXTRACT_ROT_X(r) ((r & 0x3FF0) << 2)
+#define EXTRACT_ROT_Y(r1, r2) ((((r1 & 0xF) << 6) | ((r2 & 0xFC00) >> 10)) << 6)
+#define EXTRACT_ROT_Z(r) ((r & 0x3FF) << 6)
```

Comment:
Let's forgo these macros and put the code directly.

Reply:
It's kinda nice to have the nice defined names though. Because that code is ugly and dense to read. 😆

Reply:
I've create a `M_ExtractRotation` function. For TR2 rots with two values, we will still need to utilise this but its words are reversed so a function will avoid repetition when the time comes.